project/Build.scala

```diff
@@ -930,7 +930,8 @@ object Build {
 
     /** Depends on the sources of another project. */
     def dependsOnSource(dependency: MultiScalaProject): MultiScalaProject = {
-      if (isGeneratingForIDE) project.dependsOn(dependency)
+      if (isGeneratingForIDE && !project.shouldDependsOnSourceInIDE)
```

Comment:
```suggestion
      if (isGeneratingForIDE && !project.dependsOnSourceInIDE)
```
This would probably read better.

Reply:
thanks! done https://github.com/scala-native/scala-native/pull/3450/commits/0b305f4810ca3ee31e0cb51cfd94bb2ef441e845